### ISSUE TITLE
SwaggerJsonTransformer seems to remove unreferenced type

### ISSUE THREAD

**Issue by DaKaLKa** [Burgyn/MMLib.SwaggerForOcelot#128]
To generate my swagger.json (OpenAPI), I'm using swashbuckle with some IDocumentFilter that add some additional types to the swagger.json (types which are not used as parameter in any endpoint). 
The original swagger.json (OpenAPI) is correct and swagger-ui is displaying all types.

The swagger.json and the ui that is provided by SwaggerForOcelot seems to remove some "component.schemas"-entries. 

**Expected behavior**
All schemas are in transformed json-file, not only referenced

**To Reproduce**
1. Extract the "test.json" file from attached "test.zip"-File and add it to SwaggerEndPoints
2. TestType is removed from "components.schemas": 
In "test.json":
![image](https://user-images.githubusercontent.com/6772531/94553106-46045a00-0258-11eb-91c7-9d531052f2e2.png)
After json-Transformation:
![image](https://user-images.githubusercontent.com/6772531/94553198-6e8c5400-0258-11eb-92d0-78919ba860c5.png)
[](url)

Zip-File containing "test.json":
[test.zip](https://github.com/Burgyn/MMLib.SwaggerForOcelot/files/5298638/test.zip)

**Comment by Burgyn** [comment]
Hi,

Thank you for your issue. By design, this package removes unused types from the output swagger json.
If you want, you can set `"TransformByOcelotConfig": false` on `SwaggerEndpoint`. However, this means that the source swagger will not be transformed at all.

---
I can imagine that I will support setting whether unused types should be removed, but this is not the case at the moment.

**Comment by alanag13** [comment]
@Burgyn This is a problem for types that are derived from other types (via `allOf: [{"$ref": "#/definitions/path"}, {...other props...}]`).
In this case, only `other props` seem to be considered, and the ref'd types are considered "unused" and eliminated, resulting in invalid swagger.

For example, the below swagger, when transformed, would remove the `Bar` type, although it is referenced and needed by the `Foo` type. It seems as if the type is considered "unused" if there are no paths that have a `schema` that directly reference it, but per below, this isn't always the case.

```json
{
  "swagger": "2.0",
  "info": {
    "title": "Test",
    "version": "1",
  },
  "paths": {
    "/api/v1/foo": {
      "get": {
        "responses": {
        "200": {
          "x-nullable": false,
          "description": "Success.",
          "schema": {
            "$ref": "#/definitions/Foo"
          }
        }
      } 
      }
    }
    },
    "definitions": {
      "Foo": {
        "allOf": [
          {
            "$ref": "#/definitions/Bar"
          },
          {
            "type": "object",
            "properties": {
              "propTest": {
                "type": "string"
              }
            }
          }
        ]
      },
      "Bar": {
        "type": "object",
        "properties": {
          "anotherProp": {
            "type": "string"
          }
        }
      }
    }
  }
```

**Comment by alanag13** [comment]
I made a quick reproduction of this here:
https://dotnetfiddle.net/zEQV4N

You can see that the `Bar` type is not output.

**Comment by alanag13** [comment]
@Burgyn I'm realizing that what I'm describing above is most likely a bug and different than the "by design" choice to remove types that are _truly_ unused -- let me know if I should create a separate issue for this.

**Comment by Burgyn** [comment]
Hi @alanag13,
I'll look into it tomorrow or the day after tomorrow. Sorry, I'm pretty busy right now.

**Comment by Burgyn** [comment]
Hi @alanag13,

thanks for your report and effort which help me to reproduce it.

This bug is fixed by PR #132 and now is publishing to nuget as version `v2.1.6`.

**Comment by alanag13** [comment]
Thanks @Burgyn ! I'll try it out right away.
